server/api/runtime_handlers/sparkjob/spark3job.py

```diff
@@ -694,5 +702,35 @@ def _enrich_job(
             )
         return
 
+    def _enrich_node_selectors_with_project_level(self, runtime, job, project_name):
+        # After adding the runtime executor and driver node selectors to the job object,
+        # we enrich the node selectors with any additional selectors from the project-level configuration, if exists.
+        if runtime._get_db() and project_name:
+            project = runtime._get_db().get_project(project_name)
+            if project.spec.default_function_node_selector:
+                if runtime.spec.driver_node_selector:
+                    update_in(
+                        job,
+                        "spec.driver.nodeSelector",
+                        mlrun.utils.helpers.select_non_empty_fields(
+                            {
+                                **project.spec.default_function_node_selector,
+                                **runtime.spec.driver_node_selector,
+                            }
+                        ),
+                    )
+                if runtime.spec.executor_node_selector:
+                    update_in(
+                        job,
+                        "spec.executor.nodeSelector",
+                        mlrun.utils.helpers.select_non_empty_fields(
+                            {
+                                **project.spec.default_function_node_selector,
+                                **runtime.spec.executor_node_selector,
+                            }
+                        ),
+                    )
```

Comment:
And what if we don't have neither of them (executor/driver selectors)?
don't we still want to enrich with project node selectors?

Reply:
it is being enriched prior to this function